### PR TITLE
Update the CreateNewAzureContainer MSBuild task to support authentication via Azure CLI

### PR DESCRIPTION
**Pull request by andriipatsula**
* Added the `Azure.Identity` dependency.
* Extended AzureStorageUtils with additional constructor.
* Updated CreateNewAzureContainer, CreateAzureContainerIfNotExists.

### To double check:

* [ ] The right tests are in and the right validation has happened.  Guidance: https://github.com/dotnet/arcade/blob/main/Documentation/Validation.md

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs [hunk 1]
```diff
@@ -66,12 +63,20 @@ public enum PackageFeedStatus
...
+            StorageSharedKeyCredential credential = new(AccountName, AccountKey);
             Uri endpoint = new Uri($"https://{AccountName}.blob.core.windows.net");
-            BlobServiceClient service = new BlobServiceClient(endpoint, _credential, s_clientOptions);
+            BlobServiceClient service = new BlobServiceClient(endpoint, credential, s_clientOptions);
             Container = service.GetBlobContainerClient(ContainerName);
         }
 
+        public AzureStorageUtils(string accountName, TokenCredential credential, string containerName)
+        {
+            Uri endpoint = new Uri($"https://{accountName}.blob.core.windows.net");
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs:74]: nit 
```suggestion
            Uri endpoint = new($"https://{accountName}.blob.core.windows.net");
```

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs [hunk 2]
```diff
@@ -66,12 +63,20 @@ public enum PackageFeedStatus
 
         public AzureStorageUtils(string AccountName, string AccountKey, string ContainerName)
         {
-            _credential = new StorageSharedKeyCredential(AccountName, AccountKey);
+            StorageSharedKeyCredential credential = new(AccountName, AccountKey);
             Uri endpoint = new Uri($"https://{AccountName}.blob.core.windows.net");
-            BlobServiceClient service = new BlobServiceClient(endpoint, _credential, s_clientOptions);
+            BlobServiceClient service = new BlobServiceClient(endpoint, credential, s_clientOptions);
```
Comment (dkurepa) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs:68]:
```suggestion
            BlobServiceClient service = new(endpoint, credential, s_clientOptions);
```

Comment (andriipatsula) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs:68]: Updated. Thanks

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs (hunk 1 above)
Comment (andriipatsula) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs:74]: Updated. Thanks

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs [hunk 3]
```diff
@@ -66,12 +63,20 @@ public enum PackageFeedStatus
...
             Container = service.GetBlobContainerClient(ContainerName);
         }
 
+        public AzureStorageUtils(string accountName, TokenCredential credential, string containerName)
+        {
+            Uri endpoint = new Uri($"https://{accountName}.blob.core.windows.net");
+            BlobServiceClient service = new BlobServiceClient(endpoint, credential, s_clientOptions);
```
[the remark this answers is not in the document]
Comment (andriipatsula) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/common/AzureStorageUtils.cs:75]: Updated. Thanks

**Review by dkurepa — APPROVED** [review]

**Comment by MilenaHristova** [comment]
\backport to release/8.0

**Pull request merged**